pkg/storegateway/series_refs.go

```diff
@@ -766,18 +766,25 @@ func (s *loadingSeriesChunkRefsSetIterator) Next() bool {
 			s.err = errors.Wrap(err, "read series")
 			return false
 		}
-		if lset.Len() == 0 {
-			// No matching chunks for this time duration, skip series
+		if len(lset) == 0 {
+			// This was a series without any chunks; we skip it
 			continue
 		}
-
+		var groups []seriesChunkRefsGroup
+		if !s.skipChunks {
+			groups = partitionChunks(chks, s.blockID)
```

Comment:
I think the partitioning into groups is a complication that we may avoid. Let's discuss vis-a-vis about it.